src/InputSource.h

```diff
@@ -7,39 +7,28 @@
 #include <Halide.h>
 #include "finish.h"
 
-class AbstractInputSource {
```

Comment:
I decided to simplify IO because there is only one type of inputs - RAW files, so this abstraction is redundant.